nest-tax-backend/src/utils/middlewares/logger.ts

```diff
@@ -19,34 +23,47 @@ export class AppLoggerMiddleware implements NestMiddleware {
           ?.toString()
           .includes('application/json')
       ) {
-        responseData = exitData.toString().slice(0, 1000)
+        // Filter out keys starting with `$`. We will log them later
+        const { responseLog, responseMessage } = separateLogFromResponseObj(
+          typeof exitData === 'string' ? JSON.parse(exitData) : exitData,
+        )
+        // eslint-disable-next-line no-param-reassign
+        exitData = JSON.stringify(responseMessage)
```

Comment:
Please avoid modifying existing variables. Especially arguments of the function, please create a new variable each time something happens.